### PR TITLE
Add logutil package with log writer wrapper

### PR DESCRIPTION
**Pull request by andrewsun2898**
The logutil package will contain a writer wrapper implementation that
creates a file for logging purposes. It will use a limited writer to
ensure that the log doesn't grow unboundedly.

Signed-off-by: Andrew <andrewsun@google.com>

### CODE REVIEW

File: pkg/logutil/logutil.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"github.com/nanmu42/limitio"
+)
+
+// NewWriterToFile creates a Writer that writes out output to a file path, creating necessary directories if needed, up to a maximum limit maxSize.
+func NewWriterToFile(maxSize int, dirPath, name string) (io.Writer, error) {
+	if err := os.MkdirAll(dirPath, 0700); err != nil {
+		return nil, err
+	}
+
+	logFile, err := os.OpenFile(dirPath+"/"+name, os.O_CREATE|os.O_APPEND|os.O_RDWR, 0666)
```
Comment (10000TB) [inline pkg/logutil/logutil.go:21]: consider `filepath.Join`

File: pkg/logutil/logutil.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"io"
+	"os"
+
+	"github.com/nanmu42/limitio"
+)
+
+// NewWriterToFile creates a Writer that writes out output to a file path, creating necessary directories if needed, up to a maximum limit maxSize.
+func NewWriterToFile(maxSize int, dirPath, name string) (io.Writer, error) {
+	if err := os.MkdirAll(dirPath, 0700); err != nil {
```
Comment (10000TB) [inline pkg/logutil/logutil.go:17]: consider removing this logic, to create direction if it is missing, and ask that caller to make sure right directory is passed in.

One advantage is we not get in the case, that if an unintended directory is passed in, we don't end up create a new directory, and store logs there.

File: pkg/logutil/logutil_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		},
+		{
+			name:          "dir missing",
+			dirPath:       "dir",
+			filename:      "file.log",
+			maxSize:       1024,
+			startContent:  []byte(""),
+			appendContent: []byte("bar"),
+			wantContent:   []byte("bar"),
+		},
```
Comment (10000TB) [inline pkg/logutil/logutil_test.go:58]: add test cases for covering line 23, and 28

**Review by 10000TB — CHANGES_REQUESTED** [review]

File: pkg/logutil/logutil.go (hunk 1 above)
Comment (andrewsun2898) [reply pkg/logutil/logutil.go]: done.

File: pkg/logutil/logutil.go (hunk 2 above)
Comment (andrewsun2898) [reply pkg/logutil/logutil.go]: good point, done.

File: pkg/logutil/logutil_test.go (hunk 3 above)
Comment (andrewsun2898) [reply pkg/logutil/logutil_test.go:64]: Not sure how to test line 28.. somehow make the stat fail?

Comment (10000TB) [reply pkg/logutil/logutil_test.go:64]: Worth checking implementation of Stat, and see which case will it return an error, and mock accordingly.

**Review by 10000TB — APPROVED** [review]

**Pull request merged**